# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/MozContainerWayland.cpp
# Commit: c28a3faf7769
# Full Hash: c28a3faf7769cc9eb407418cc728dd322fe9b78a
# Author: stransky <stransky@redhat.com>
# Date: 2022-12-15 16:30:22
# Description:
#   Bug 1804973 [Wayland] Check size for valid EGLWindows only r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164651
# ==============================================================================

diff -r 666b29472d99 -r c28a3faf7769 widget/gtk/MozContainerWayland.cpp
--- a/widget/gtk/MozContainerWayland.cpp	Thu Dec 15 08:07:32 2022 +0000
+++ b/widget/gtk/MozContainerWayland.cpp	Thu Dec 15 08:09:22 2022 +0000
@@ -704,6 +704,9 @@
                                                         nsIntSize aSize,
                                                         int aScale) {
   MozContainerWayland* wl_container = &container->wl_container;
+  if (!wl_container->eglwindow) {
+    return false;
+  }
   if (wl_container->buffer_scale != aScale) {
     return true;
   }
